controlpanel/api/helm.py

```diff
@@ -0,0 +1,150 @@
+from datetime import datetime
+import logging
+import re
+import subprocess
+
+from django.conf import settings
+from rest_framework.exceptions import APIException
+
+
+log = logging.getLogger(__name__)
+
+
+class HelmError(APIException):
+    status_code = 500
+    default_code = "helm_error"
+    default_detail = "Error executing Helm command"
+
+
+class Helm(object):
+
+    def __init__(self):
+        self.enabled = settings.ENABLED['write_to_cluster']
+
+    def _execute(self, *args, check=True, **kwargs):
+        if not self.enabled:
+            return None
+
+        should_wait = False
+        if 'timeout' in kwargs:
+            should_wait = True
+            timeout = kwargs.pop('timeout')
+
+        try:
+            proc = subprocess.Popen(
+                ["helm", *args],
+                stderr=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+                encoding='utf8',
+                **kwargs,
+            )
```

Comment:
### Refactor Helm client module

> Lines 1 - 88

I simplified the Helm client code, which had two wrapper methods around `subprocess` calls. Instead, we have a single method to execute Helm and potentially run an asynchronous process for later use in Server Sent Events.

[Next →]()